servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -0,0 +1,648 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.block;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.thrift.TProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tachyon.Constants;
+import tachyon.HeartbeatExecutor;
+import tachyon.HeartbeatThread;
+import tachyon.StorageDirId;
+import tachyon.StorageLevelAlias;
+import tachyon.conf.TachyonConf;
+import tachyon.master.IndexedSet;
+import tachyon.master.MasterBase;
+import tachyon.master.block.journal.BlockIdGeneratorEntry;
+import tachyon.master.block.journal.BlockInfoEntry;
+import tachyon.master.block.journal.WorkerIdGeneratorEntry;
+import tachyon.master.block.meta.MasterBlockInfo;
+import tachyon.master.block.meta.MasterBlockLocation;
+import tachyon.master.block.meta.MasterWorkerInfo;
+import tachyon.master.journal.Journal;
+import tachyon.master.journal.JournalEntry;
+import tachyon.master.journal.JournalInputStream;
+import tachyon.master.journal.JournalOutputStream;
+import tachyon.thrift.BlockInfo;
+import tachyon.thrift.BlockInfoException;
+import tachyon.thrift.BlockLocation;
+import tachyon.thrift.BlockMasterService;
+import tachyon.thrift.Command;
+import tachyon.thrift.CommandType;
+import tachyon.thrift.NetAddress;
+import tachyon.thrift.WorkerInfo;
+import tachyon.util.CommonUtils;
+import tachyon.util.FormatUtils;
+import tachyon.util.ThreadFactoryUtils;
+import tachyon.util.io.PathUtils;
+
+/**
+ * This master manages the metadata for all the blocks and block workers in Tachyon.
+ */
+public final class BlockMaster extends MasterBase implements ContainerIdGenerator {
```

Comment:
It seems `BlockMaster` is the only one implementing `ContainerIdGenerator`, then we don't we just always refer to `BlockMaster` in all occurrences where they need  `ContainerIdGenerator`?